feature/experimental/p4rt/ate_tests/traceroute_packetout_test/packetout_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceroute_packetout_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/cisco-open/go-p4/p4rt_client"
+	"github.com/openconfig/ondatra"
+	p4v1 "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+type PacketIO interface {
+	GetPacketOut(portID uint32, isIPv4 bool, ttl uint8) []*p4v1.PacketOut
+}
+
+type testArgs struct {
+	ctx      context.Context
+	leader   *p4rt_client.P4RTClient
+	dut      *ondatra.DUTDevice
+	ate      *ondatra.ATEDevice
+	top      *ondatra.ATETopology
+	packetIO PacketIO
+}
+
+// sendPackets sends out packets via PacketOut message in StreamChannel.
+func sendPackets(t *testing.T, client *p4rt_client.P4RTClient, packets []*p4v1.PacketOut, packetCount int) {
+	count := packetCount / len(packets)
+	for _, packet := range packets {
+		for i := 0; i < count; i++ {
+			if err := client.StreamChannelSendMsg(
+				&streamName, &p4v1.StreamMessageRequest{
+					Update: &p4v1.StreamMessageRequest_Packet{
+						Packet: packet,
+					},
+				}); err != nil {
+				t.Errorf("There is error seen in Packet Out. %v, %s", err, err)
+
+			}
+		}
+	}
+}
+
+// testPacketOut sends out PacketOut with payload on p4rt leader or
+// follower client, then verify DUT interface statistics
+func testPacketOut(ctx context.Context, t *testing.T, args *testArgs) {
+	leader := args.leader
+	desc := "PacketOut from Primary Controller"
+	ttls := []int{0, 1}
+	//for ipv4
+	t.Run(desc, func(t *testing.T) {
```

Comment:
Sorry about the confusion, I meant to distinguish the desc in this and the one below for ipv6
`	t.Run(desc + " ipv4", func(t *testing.T) {`

Reply:
Sure. I have made the change